proofread: sync_accounts and upgrade (#2511)
juicedata/juicefs: docs/en/administration/sync_accounts_between_multiple_hosts.md

# Sync Accounts between Multiple Hosts

JuiceFS supports POSIX compatible ACL to manage permissions by the granularity of directory or file. The behavior is the same as a local file system.

To provide users with an intuitive and consistent permission management experience (e.g. the files accessible by user A on host X should be accessible by the same user on host Y), the same user who wants to access JuiceFS should have the same UID and GID on all hosts.

Here we provide a simple [Ansible](https://www.ansible.com/community) playbook to demonstrate how to ensure an account with same UID and GID on multiple hosts.

:::note
If you are using JuiceFS in Hadoop environment, besides sync accounts between multiple hosts, you can also specify a global user list and user group file. Please refer to [here](../deployment/hadoop_java_sdk.md#other-configurations) for more information.
:::

## Install Ansible

> Select a host as a [control node](https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html#managed-node-requirements) which can access all hosts using `ssh` with the same privileged account like `root` or other sudo account. Then, install Ansible on this host. Read [Installing Ansible](https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html#installing-ansible) for more installation details.

Reviewer: too much "install" in one sentence, suggestion:

for more installation details. -> for details.

Author: ```suggestion
Select a host as a [control node](https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html#managed-node-requirements) which can access all hosts using `ssh` with the same privileged account like `root` or other sudo account. Then, install Ansible on this host. Read [Installing Ansible](https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html#installing-ansible) for details.
```

## Ensure the same account on all hosts

> Create an empty directory `account-sync`, and save the following content in `play.yaml` under this directory.

Reviewer: Create `account-sync/play.yaml` as follows:

> Create an empty directory `account-sync`, and save the following content in `play.yaml` under this directory.

Author: ```suggestion
Create `account-sync/play.yaml` as follows:
```

> Create a file named `hosts` in this directory. Place IP addresses of all the hosts that need to create account in this file with one host's IP per line.

Reviewer: I think it'd be better with:

> Create the Ansible inventory file `hosts`, which contains hostname / IP address for all hosts that need to create account.

there're examples below, I won't worry too much about the "IP per line" part.

> Create a file named `hosts` in this directory. Place IP addresses of all the hosts that need to create account in this file with one host's IP per line.

Author: ```suggestion
Create the Ansible inventory `hosts`, which contains IP addresses of all hosts that need to create account.
```

172.16.255.180             : ok=1    changed=0    unreachable=0    failed=1
```

> In the above example, the group ID 1000 has been allocated to another group on host `172.16.255.180`. So we should **change the GID**  or **delete the group with GID 1000** on host `172.16.255.180`, and then run the playbook again.

Reviewer: two consecutive space character in this line

:::caution

> If a user account has already existed on the host and we change it to another UID or GID value, the user may lose permissions to the files and directories which they previously had. For example:

Reviewer: If UID / GID has been changed on an existing user, the user may lose permissions to previously accessible files.

Author: ```suggestion
If the UID / GID of an existing user is changed, the user may lose permissions to previously accessible files. For example:
```
